locallib.php

```diff
@@ -355,25 +355,27 @@ function accredible_get_recipient_sso_linik($group_id, $email) {
 /**
  * List all of the issuer's templates
  *
+ * @param apiRest $apiRest
  * @return array[stdClass] $templates
  */
-function accredible_get_templates() {
-    global $CFG;
-
-    $curl = curl_init(get_api_endpoint().'issuer/templates');
-    curl_setopt( $curl, CURLOPT_HTTPHEADER, array( 'Authorization: Token token="'.$CFG->accredible_api_key.'"' ) );
-    curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
-    if (!$result = json_decode( curl_exec($curl) )) {
+function accredible_get_templates($apiRest = null) {
+    // An apiRest with a mock client is passed when unit testing.
+    if(!$apiRest) {
+        $apiRest = new apiRest();
+    }
+    $response = $apiRest->search_groups(10000, 1);
```

Comment:
Ideally, it should take care of the pagination, but this is a part of the legacy feature so I chose to stick with the page size `10000` as `accredible_get_groups()` in locallib.php.